### PR TITLE
feat: add tracking to staking

### PR DESCRIPTION
**Pull request by rosepuppy**
Added tracking to StakingForm, UnstakingForm, and StakingRewardDialog to track on submit transactions. 

Also refactored the AnalyticsEventData type to be more readable.

### CODE REVIEW

File: src/views/forms/UnstakeForm/index.tsx [hunk 1]
```diff
@@ -133,10 +143,14 @@ export const UnstakeForm = ({ onDone, className }: UnstakeFormProps) => {
     } finally {
       setIsLoading(false);
     }
-  }, [isAmountValid, amounts, undelegate, onDone]);
+  }, [isAmountValid, amounts, undelegate, onDone, totalAmount]);
 
   const onChangeAmount = useCallback((validator: string, value: number | undefined) => {
     setAmounts((a) => ({ ...a, [validator]: value }));
+    track(AnalyticsEvent.UnstakeInput, {
```
Comment (moo-onthelawn) [inline src/views/forms/UnstakeForm/index.tsx:150]: hmm I wonder if we should debounce the change amount if we're logging here

File: src/views/forms/UnstakeForm/index.tsx [hunk 2]
```diff
@@ -121,7 +124,14 @@ export const UnstakeForm = ({ onDone, className }: UnstakeFormProps) => {
...
     try {
       setIsLoading(true);
-      await undelegate(amounts);
+      const tx = await undelegate(amounts);
+      const txHash = hashFromTx(tx.hash);
+
+      track(AnalyticsEvent.UnstakeTransaction, {
+        txHash,
+        amount: totalAmount,
+        validatorAddresses: Object.keys(amounts),
```
Comment (moo-onthelawn) [inline src/views/forms/UnstakeForm/index.tsx:133]: would it be worth also incorporating whether the transaction was successful or failed? (i.e. by tracking after onDone and in the catchError?)

File: src/views/forms/StakeForm/index.tsx [hunk 3]
```diff
@@ -115,7 +122,14 @@ export const StakeForm = ({ onDone, className }: StakeFormProps) => {
     }
     try {
       setIsLoading(true);
-      await delegate(selectedValidator.operatorAddress, amountBN.toNumber());
+      const tx = await delegate(selectedValidator.operatorAddress, amountBN.toNumber());
+      const txHash = hashFromTx(tx.hash);
+
+      track(AnalyticsEvent.StakeTransaction, {
+        txHash,
+        amount: amountBN.toNumber(),
```
Comment (moo-onthelawn) [inline src/views/forms/StakeForm/index.tsx:130]: similar comment as below

File: src/views/dialogs/StakingRewardDialog.tsx [hunk 4]
```diff
@@ -117,7 +120,13 @@ export const StakingRewardDialog = ({ validators, usdcRewards, setIsOpen }: Elem
     try {
       setIsLoading(true);
       setError(undefined);
-      await withdrawReward(validators);
+      const tx = await withdrawReward(validators);
+      const txHash = hashFromTx(tx.hash);
+
+      track(AnalyticsEvent.ClaimTransaction, {
+        txHash,
+        amount: usdcRewards.toString(),
```
Comment (moo-onthelawn) [inline src/views/dialogs/StakingRewardDialog.tsx:128]: similar comment to whether we want to track success

File: src/views/forms/StakeForm/index.tsx [hunk 5]
```diff
@@ -107,6 +110,10 @@ export const StakeForm = ({ onDone, className }: StakeFormProps) => {
 
   const onChangeAmount = (value?: BigNumber) => {
     setAmountBN(value);
+    track(AnalyticsEvent.StakeInput, {
```
Comment (moo-onthelawn) [inline src/views/forms/StakeForm/index.tsx:113]: simialr comment about debouncing

**Review by moo-onthelawn — APPROVED** [review]
some considers!

File: src/views/forms/UnstakeForm/index.tsx (hunk 1 above)
Comment (rosepuppy) [reply src/views/forms/UnstakeForm/index.tsx:150]: I spoke to Yang about it when I added this, and he was ok with tracking it like this. But I'll look into debouncing the tracking specifically

File: src/views/forms/UnstakeForm/index.tsx (hunk 2 above)
Comment (yang-dydx) [reply src/views/forms/UnstakeForm/index.tsx:134]: Should we track on "click"? This is just tracking when it's done right since we get the txn hash

**Pull request merged**